src/components/Viewer.tsx

```diff
@@ -27,11 +28,66 @@ function WrappedViewStateDeck({ layers }: { layers: Layer<VivLayerProps>[] }): J
 }
 
 function Viewer(): JSX.Element {
+
+  const sourceInfo = useRecoilValue(sourceInfoState);
+
+  // Click handler for plate layout - opens image in new window
+  const handleClick = (info: Object) => {
+    let layerId = (info as any).layer.id;
+    if (!layerId.includes('-plate-')){
+      return;
+    }
+    // Get the info we need from the layerId
+    const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+    const layerPropsId = layerId.split('-plate-')[0];
+    const [row, col] = layerId.split('-plate-')[1].split('-').map((x: string) => parseInt(x));
+    let source = (info as any).sourceLayer?.props?.source;
+    const plateAcquisitions = sourceInfo[layerPropsId]?.plateAcquisitions;
+    if (source && !isNaN(row) && !isNaN(col) && plateAcquisitions) {
+      if (source.endsWith('/')){
+        source = source.slice(0, -1);
+      }
+      let imgSource = `${source}/${plateAcquisitions[0]}/${letters[row]}/${col + 1}/Field_1/`;
+      window.open(window.location.origin + '?source=' + imgSource);
+    }
+  }
+
   const layerConstructors = useRecoilValue(layersSelector);
-  const layers = layerConstructors.map((l) => {
+  const layers = layerConstructors.flatMap((l) => {
     // Something weird with Recoil Loadable here. Need to cast to any.
     const { Layer, layerProps, on } = l as any;
-    return !Layer || !on ? null : new Layer(layerProps);
+    const layerInfo = sourceInfo[layerProps.id];
+    const rows = layerInfo?.rows;
+    const columns = layerInfo?.columns;
+    const loaders = layerInfo.loaders;
+    if (rows && columns && layerProps.loader && loaders) {
+      const [height, width] = layerProps.loader.base.shape.slice(-2);
+      const spacer = 5;
+      const top = -(rows * (height + spacer)) / 2;
+      const left = -(columns * (width + spacer)) / 2;
+      return range(rows).flatMap((row) => {
+        return range(columns).flatMap((col) => {
+          const y = top + (row * (height + spacer));
+          const x = left + (col * (width + spacer));
+          // NB: this ID is used by onClick to get row and col
+          const id = `${layerProps.id}-plate-${row}-${col}`;
+          const loader = layerInfo.loaders ? layerInfo.loaders[col + (row * columns)]: undefined;
+          if (!loader) {
+            return [];
+          }
+          let wellProps = {
+            ...layerProps,
+            id,
+            translate: [x, y],
+            loader: layerInfo.loaders ? layerInfo.loaders[col + (row * columns)] : null,
+            pickable: true,
+            onClick: handleClick
```

Comment:
It will very useful if we can override the default click handler for the ImJoy API, this will allow handle the click in other plugins in Python. Or use it as a plat explorer and when the user click on one we load the image into [ImageJ.JS](https://ij.imjoy.io).

Perhaps something like this:
 ```suggestion
            onClick: sourceInfo.click_callback || handleClick
```